cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -200,54 +200,47 @@ func (s *Server) GetAvailablePackageSummaries(ctx context.Context, request *core
 		return nil, status.Errorf(codes.InvalidArgument, "Unable to intepret page token %q: %v", request.GetPaginationOptions().GetPageToken(), err)
 	}
 
-	// A GetAvailablePackageSummaries request with pageSize=-1 will return a GetAvailablePackageSummariesResponse
-	// with a single empty AvailablePackageSummary containing only the distinct category names considering
-	// the FilterOptions. Otherwise, return the usual response after querying all the available packages
-	if pageSize == -1 {
-		chartCategories, err := s.manager.GetAllChartCategories(cq)
-		if err != nil {
-			return nil, status.Errorf(codes.Internal, "Unable to fetch chart categories: %v", err)
-		}
+	// This plugin will include, as part of the GetAvailablePackageSummariesResponse,
+	// a "Categories" field containing only the distinct category names considering the FilterOptions
```

Comment:
So I know that the existing code accepts a chart query argument, that you've used below. But in the UX, I see that we always show all categories for the namespace, regardless of any filtering. Which at first left me confused as to why we'd want to return just the categories for the packages (charts) matching the query here. But then I realised that the UX can just use the categories from the initial response when the user views the catalog? Is that what you intend?

Reply:
As it is currently implemented in the UI, we do make two separate calls. In fact, this could be an issue in that a size=0 query will return every availablePackageSummary... . We might work around it by setting size=1, but still a non-elegant solution.
I can try reducing it to a single call in https://github.com/kubeapps/kubeapps/pull/3203, once we got the first UI PR merged.

Allowing passing the FilterOptions through is just for not limiting any future use case; I mean, what if we decide to make the categories dynamic? Another reason is that we are already using a "chartQuery" when passing the namespace, so why not passing also the rest of the query?

Reply:
> We might work around it by setting size=1, but still a non-elegant solution.

But what I'm saying is that I don't think we'd need to, in that, when a user first navigates to the "App Catalog", the UI will be calling `GetAvailablePackageSummaries()` with some pagination options only (ie. no filtering). The `utils.ChartQuery` struct only will only have the namespace set. So this initial response will contain all the categories that you need, I think? Or why do you think a second query would be required with size=1 or similar to get them? The only can I can imagine that being necessary is if the user filters by name, then hits re-load, so the UX redux state does not already contain all the categories for the namespace or similar. Or maybe we don't (yet) store them in the redux state.